Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -1261,4 +1263,9 @@ static void webProcessPoolHighDynamicRangeDidChangeCallback(CMNotificationCenter
 }
 #endif // PLATFORM(MAC) || PLATFORM(IOS)
 
+void WebProcessPool::notifyMediaNetworkingActivity(bool activity)
+{
+    [[NSNotificationCenter defaultCenter] postNotificationName:WebKitMediaNetworkingActivityNotification object:[NSNumber numberWithBool:activity]];
```

Comment:
Why this mechanism over per WKWebView regular delegates?
If the goal is to send this info to network process, why not directly sending it to network process?

Reply:
because the mechanism to send current media state to the UI process was already existing.
The UI process already tracks the media state of all content processes and all pages, so re-using the infrastructure made a lot of sense.

The listener of this notification isn't the network process per say, but the radio stack on iPhone.
So having the UI process emit the notification over the network process is a non-issue

Reply:
Also the "notification-as-API" is not something we chose, it's something we have to use.